### PR TITLE
paho-mqtt-c: fix cmake names

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **paho-mqtt-c/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

targets are exported since version 1.3.2
See https://github.com/eclipse/paho.mqtt.c/blob/3148fe2d5f4b87e16266dfe559c0764e16ca0546/src/CMakeLists.txt#L308 and check targets added to `eclipse-paho-mqtt-cTargets` EXPORT

### CODE REVIEW

File: recipes/paho-mqtt-c/all/conanfile.py [hunk 1]
```diff
@@ -87,21 +87,35 @@ def package(self):
...
+            self.cpp_info.components["_paho-mqtt-c"].requires = ["openssl::openssl"]
+
+    @property
+    def _cmake_target(self):
+        target = "paho-mqtt3"
+        target += "a" if self.options.asynchronous else "c"
+        if self.options.ssl:
+            target += "s"
+        if not self.options.shared:
+            target += "-static"
```
Comment (gocarlos) [inline recipes/paho-mqtt-c/all/conanfile.py:120]: poh, this is a mess :woman_dancing:

Comment (gocarlos) [reply recipes/paho-mqtt-c/all/conanfile.py:120]: cant we just change this upstream?

what do you think? @fpagliughi
could we harmonize the paho-mqtt-c and paho-mqtt-cpp to have sane targets and find cmake names

Comment (gocarlos) [reply recipes/paho-mqtt-c/all/conanfile.py:120]: I propose `PahoMqttC` and `PahoMqttCpp` for find cmake names and `paho::mqtt-c` and `paho::mqtt-cpp`
we could also replace PAHO with ECLIPSE...

Comment (gocarlos) [reply recipes/paho-mqtt-c/all/conanfile.py:120]: and remove all the madness with static, shared, ssl etc. changes to the name 

I mean we could also change the name based on zlib, xy, brotli etc. or use another crypto lib like mbedtls and also add this to the name -> becomes unmaintainable

Comment (SSE4) [reply recipes/paho-mqtt-c/all/conanfile.py:120]: this is okay to provide different target names for different flavours of libraries.
e.g. in conan, we usually don't build static and shared in the same package. but, not everyone uses conan...
while without conan, it's pretty common practice to allow to build shared and static simultaneously (or another flavors, e.g. SSL on + SSL off, debug + release, 32 + 64, etc).
in this case, consumer has to somehow distinguish such flavours, and the best way is that he can refer to these libraries by different CMake targets - that's the most straightforward and obvious way.
so IMO there is nothing to be fixed upstream - just follow the upstream's naming in the recipe, doesn't matter how complicated it is.

Comment (gocarlos) [reply recipes/paho-mqtt-c/all/conanfile.py:120]: @SSE4 interesting that you support this :D

I guess you know the needed effort to support conan recipes across multiple years if everyone starts creating special options, names etc.

**Review by danimtb — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Comment by keysight-daryl** [comment]
I think it would be good idea to add the upstream paho-mqtt-c author @icraggs on all PR reviews for this package since he is now active on here developing this.

File: recipes/paho-mqtt-c/all/conanfile.py (hunk 1 above)
Comment (keysight-daryl) [reply recipes/paho-mqtt-c/all/conanfile.py:120]: Thing is he's right that not everyone uses conan and there are plenty of cases where builds generate multiple flavors and need different names.  Unfortunate but true.

**Comment by keysight-daryl** [comment]
upstream is in the process of releasing the next version of this package right now and will need to reconcile with all these changes shortly.

**Comment by uilianries** [comment]
Great news! Thanks @keysight-daryl !

**Pull request merged**